lib/src/device_connector.dart

```diff
@@ -118,16 +102,18 @@ class DeviceConnector {
         connectionTimeout: connectionTimeout,
       );
     } else {
-      final scanSubscription = _scanForDevices(
+      final scanSubscription = _deviceScanner
+          .scanForDevices(
               withServices: withServices, scanMode: ScanMode.lowLatency)
           .listen((DiscoveredDevice scanData) {}, onError: (Object _) {});
       Future<void>.delayed(prescanDuration).then<void>((_) {
         scanSubscription.cancel();
       });
 
-      return _getCurrentScan()
-          .future
-          .then((_) => true)
+      return _deviceScanner.currentScan.future
+          .then(
+            (_) => true,
+          )
```

Comment:
I doubt it make a lot of sense to split this across multiple lines.

Reply:
good one fixed it